### PR TITLE
Separated bootstraps from Maintenance and Setup

### PR DESCRIPTION
**Pull request by johnsimons**
I have separated the Setup and Maintenance modes into their own bootstrappers and hosts

### CODE REVIEW

**Comment by johnsimons** [comment]
@gbiellem have u had time to review ?

File: src/ServiceControl/SetupBootstrapper.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        public void Run(string username)
+        {
+            var configuration = new BusConfiguration();
+            configuration.AssembliesToScan(AllAssemblies.Except("ServiceControl.Plugin"));
+            configuration.EnableInstallers(username);
+
+            var containerBuilder = new ContainerBuilder();
+            var loggingSettings = new LoggingSettings(settings.ServiceName);
+            containerBuilder.RegisterInstance(loggingSettings);
+            var documentStore = new DocumentStore();
```
Comment (gbiellem) [inline src/ServiceControl/SetupBootstrapper.cs]: Should be EmbeddedDocumentStore

File: src/ServiceControl/MaintenanceBootstrapper.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    using ServiceBus.Management.Infrastructure.Settings;
+
+    public class MaintenanceBootstrapper
+    {
+        public void Run(HostArguments args)
+        {
+            var settings = new Settings(args.ServiceName);
+            var documentStore = new EmbeddableDocumentStore();
+
+            new RavenBootstrapper().StartRaven(documentStore, settings);
```
Comment (gbiellem) [inline src/ServiceControl/MaintenanceBootstrapper.cs]: Constructor is missing optional argument to tell Ravenbootstrapper this is maint mode. Without it's going to try and run as normal

Comment (johnsimons) [inline src/ServiceControl/MaintenanceBootstrapper.cs]: fixed

File: src/ServiceControl/Infrastructure/RavenDB/RavenBootstrapper.cs [hunk 3]
```diff
@@ -43,12 +43,12 @@ public void Customize(BusConfiguration configuration)
 
         public static Settings Settings { get; set; }
 
-        public void StartRaven(EmbeddableDocumentStore documentStore, Settings settings)
+        public void StartRaven(EmbeddableDocumentStore documentStore, Settings settings, bool maintenanceMode = false)
```
Comment (johnsimons) [inline src/ServiceControl/Infrastructure/RavenDB/RavenBootstrapper.cs]: removed

File: src/ServiceControl/SetupBootstrapper.cs (hunk 1 above)
Comment (johnsimons) [inline src/ServiceControl/SetupBootstrapper.cs]: fixed

**Pull request merged**